src/v/kafka/server/group_manager.cc

```diff
@@ -386,6 +386,34 @@ recovery_batch_consumer::operator()(model::record_batch batch) {
         auto [group_it, _] = st.groups.try_emplace(val.group_id, group_stm());
         group_it->second.update_prepared(batch.last_offset(), val);
 
+        return ss::make_ready_future<ss::stop_iteration>(
+          ss::stop_iteration::no);
+    } else if (batch.header().type == cluster::group_commit_tx_batch_type) {
+        vassert(
+          batch.record_count() == 1,
+          "commit batch must contain a single record");
+        auto r = batch.copy_records();
+        auto& record = *r.begin();
+        auto key_buf = record.release_key();
+        kafka::request_reader buf_reader(std::move(key_buf));
+        auto version = model::control_record_version(buf_reader.read_int16());
+
+        const auto& hdr = batch.header();
+        auto bid = model::batch_identity::from(hdr);
+
+        vassert(
+          version == group::commit_tx_record_version,
+          "unknown group inflight tx record version: {} expected: {}",
+          version,
+          group::commit_tx_record_version);
+
+        auto val_buf = record.release_value();
+        auto val = reflection::from_iobuf<group_log_commit_tx>(
+          std::move(val_buf));
+
+        auto [group_it, _] = st.groups.try_emplace(val.group_id, group_stm());
```

Comment:
should we `vassert()` the success here ?

Reply:
it's fine, the intent is to get group_stm by group_id or create and get if it's missing